llvm/include/llvm/ADT/Triple.h

```diff
@@ -102,8 +102,8 @@ class Triple {
     fpga_aocr,      // Intel FPGA: linked early image
     fpga_aocx,      // Intel FPGA: linked image
     fpga_dep,       // Intel FPGA: dependency file
-    fpga32,         // 32-bit FPGA
-    fpga64,         // 64-bit FPGA
+    fpga32,         // 32-bit Xilinx FPGA
```

Comment:
Are there any conflicts with Intel FPGA?
I assume we can have both Xilinx and Intel FPGA on the same machine.

Reply:
> Are there any conflicts with Intel FPGA?
not that I am aware of
> I assume we can have both Xilinx and Intel FPGA on the same machine.
I can't think of any reason it wouldn't be possible but I don't expect anyone has tried and I wouldn't be surprised if there was issue doing it. we have no tests where not all kernels are targeting the same device. 

the comment gave the impression that this target is generic when it isn't.
maybe it should be renamed.
also I am not sure there is any value to having fpga32 since to my understanding fpga don't have fixed bit width like traditional hardware.

Reply:
The target was named/commented generically because Intel hadn't really added their FPGA target yet as I remember, so I had some naive hope that it might lineup! Of course, I was wrong :-)

As for the 32/64 bit, it's mainly as the SPIR target we use has 32/64 bit cases  (https://github.com/triSYCL/sycl/blob/98a61101707652d38e71ebb8938e035f745133da/clang/lib/Basic/Targets.cpp#L556) and I believe this has some minor affect on the LinuxTargetInfo/OSTargetInfo as well (this is likely negligible for SPIR though). So it simplified adding it. I am no expert in the TargetInfo system, so there is without a doubt a better solution :-D

In essence feel free to rename them whatever you want and/or remove one! Perhaps the HLS team have a good naming convention.

Reply:
> also I am not sure there is any value to having fpga32 since to my understanding fpga don't have fixed bit width like traditional hardware.

I think this is related to the pointer size only. So we could have FPGA with less than GiB or more than 4GiB.
@aisoard: do we have HLS running with 32-bit addresses by the way?